cupy/_core/include/cupy/swap.cuh

```diff
@@ -0,0 +1,37 @@
+// This file is modified from thrust/detail/swap.h
```

Comment:
As mentioned, we really just need `thrust::swap`, so I didn't include other APIs exposed in `thrust/swap.h`.